traffic_ops/app/db/admin.go

```diff
@@ -255,35 +302,84 @@ func reset() {
 	dropDB()
 	createDB()
 	loadSchema()
-	migrate()
+	runMigrations()
 }
 
 func upgrade() {
-	goose(GooseUp)
+	runMigrations()
 	if !TrafficVault {
 		seed()
 		patch()
 	}
 }
 
-func migrate() {
-	goose(GooseUp)
+func runMigrations() {
+	initMigrate()
+	migratedFromGoose := false
+	if !TrafficVault {
+		_, _, versionErr := Migrate.Version()
+		if versionErr != nil {
+			if versionErr != migrate.ErrNilVersion {
+				die("Error running migrate version: " + versionErr.Error())
+			}
+			migratedFromGoose = true
+			if err := Migrate.Steps(1); err != nil {
+				die("Error migrating to Migrate from Goose: " + err.Error())
+			}
+		}
+	}
+	upErr := Migrate.Up()
+	if upErr == migrate.ErrNoChange {
+		if !migratedFromGoose {
+			println(upErr.Error())
+		}
+	} else if upErr != nil {
+		die("Error running migrate up: " + upErr.Error())
+	}
+
+}
+
+func runUp() {
+	runMigrations()
 }
 
 func down() {
-	goose(CmdDown)
+	initMigrate()
+	if err := Migrate.Steps(-1); err != nil {
+		die("Error running migrate down: " + err.Error())
+	}
 }
 
 func redo() {
-	goose(CmdRedo)
+	initMigrate()
+	{
```

Comment:
are these separate scopes necessary?

Reply:
It was a little optimization so it uses named `error` variables without having 2 `error` variables defined at once. I can remove the scopes, though.

Reply:
Scopes removed in e66a5eb281

Reply:
Now that I'm looking at it again, those variables weren't defined outside of the `if` statements anyway, so yeah, they were completely redundant.